### DISCUSSION TITLE
Hitchwiki SSO

### DISCUSSION THREAD

**Discussion by tillwenke** [Hitchwiki/hitchwiki#201] [Ideas]
@Hitchwiki/admins @Hitchwiki/devs 
using the hitchwiki accounts for SSO on other services.
e.g. sign into hitchmap.com with your hitchwiki account.

is this already possible?
would we want this?

**Comment by guaka** [comment]
what about using trustroots accounts instead? a lot more people log in there than on hitchwiki

[NIP-05](https://github.com/nostr-protocol/nips/blob/master/05.md) could be nice for this, https://github.com/trustrootsops/tr-nostr/issues/4

**Reply by tillwenke** [reply]
... and slowly migrating hitchwiki account to TR as a consequence as well?

**Reply by Henridv** [reply]
@tillwenke What would be the advantage of migrating the accounts to Trustroots? It would mean all Hitchwiki users would need to be Trustroots users, right? That personally doesn't feel right. These two services attract similar groups of people, I guess, but they're still different services.

How are people signing in the Hitchmap now?

**Reply by tillwenke** [reply]
right now there is no sign in. you can just use any name when submitting.

i agree with you. using the hitchwiki account would be the canonical solution.

probably one would just have to take a look how to set it up. I assume there are people who tried this before on a media wiki.

**Comment by guaka** [comment]
I had SSO on a bunch of mediawiki sites on the same server but it was a bit messy.

Also had openid stuff enabled on mediawiki, but openid never took off.  Nostr looks more promising to me.

I am not talking about migrating any accounts anywhere. I am proposing two orthogonal steps:
- NIP-05 verification functionality on trustroots, enabling this for anyone@trustroots.org
- log in with nostr pub and/or verified NIP-05 on hitchwiki etc

[This comment](https://github.com/nostr-protocol/nips/issues/154#issuecomment-1374522357) and the entire thread can explain this better than I can.

NIP-05 could also come from hitchwiki, wikiuser@hitchwiki.org, but this seems less interesting. A lot more people are active on trustroots, and trustroots has an existing social graph, which gives it anti-spam potential. E.g. if someone signs up with a trustroots NIP-05 we can initially immediately trust the account for edits. (Later: only if trustroots user has some form of social proof.)
